packages/api/tests/endpoints/test-granules-bulkMoveCollection.js

```diff
@@ -0,0 +1,336 @@
+  });
+  t.context.testPgCollection2 = fakeCollectionRecordFactory({
+    name: collectionName2,
+    version: collectionVersion2,
+  });
+  t.context.collectionPgModel = new CollectionPgModel();
+  const [pgCollection] = await t.context.collectionPgModel.create(
+    t.context.knex,
+    t.context.testPgCollection
+  );
+  const [pgCollection2] = await t.context.collectionPgModel.create(
+    t.context.knex,
+    t.context.testPgCollection2
+  );
+  t.context.collectionCumulusId = pgCollection.cumulus_id;
+  t.context.collectionCumulusId2 = pgCollection2.cumulus_id;
+
+  t.context.collectionId = constructCollectionId(pgCollection.name, pgCollection.version);
+  t.context.collectionId2 = constructCollectionId(pgCollection2.name, pgCollection2.version);
+});
+
+test.beforeEach(async (t) => {
+  t.context.createGranuleId = () => `${cryptoRandomString({ length: 7 })}.${cryptoRandomString({ length: 20 })}.hdf`;
+  const granuleId1 = t.context.createGranuleId();
+  const granuleId2 = t.context.createGranuleId();
+  const granuleId3 = t.context.createGranuleId();
+
+  // create fake Postgres granule records
+  t.context.fakePGGranules = [
+    fakeGranuleRecordFactory({
+      granule_id: granuleId1,
+      status: 'completed',
+      collection_cumulus_id: t.context.collectionCumulusId,
+      published: true,
+      cmr_link:
+        'https://cmr.uat.earthdata.nasa.gov/search/granules.json?concept_id=A123456789-TEST_A',
+      duration: 47.125,
+      timestamp: new Date(Date.now()),
+    }),
+    fakeGranuleRecordFactory({
+      granule_id: granuleId2,
+      status: 'failed',
+      collection_cumulus_id: t.context.collectionCumulusId,
+      duration: 52.235,
+      timestamp: new Date(Date.now()),
+    }),
+    fakeGranuleRecordFactory({
+      granule_id: granuleId3,
+      status: 'failed',
+      collection_cumulus_id: t.context.collectionCumulusId,
+      duration: 52.235,
+      timestamp: new Date(Date.now()),
+    }),
+    // granule with same granule_id as above but different collection_cumulus_id
+    fakeGranuleRecordFactory({
+      granule_id: granuleId3,
+      status: 'failed',
+      collection_cumulus_id: t.context.collectionCumulusId2,
+      duration: 52.235,
+      timestamp: new Date(Date.now()),
+    }),
+  ];
+
+  t.context.fakePGGranuleRecords = await Promise.all(
+    t.context.fakePGGranules.map((granule) =>
+      upsertGranuleWithExecutionJoinRecord({
+        knexTransaction: t.context.knex,
+        granule,
+        executionCumulusId: t.context.testExecutionCumulusId,
+        granulePgModel: t.context.granulePgModel,
+      }))
+  );
+  t.context.insertedPgGranules = t.context.fakePGGranuleRecords.flat();
+  const insertedApiGranuleTranslations = await Promise.all(
+    t.context.insertedPgGranules.map((granule) =>
+      translatePostgresGranuleToApiGranule({
+        knexOrTransaction: t.context.knex,
+        granulePgRecord: granule,
+      }))
+  );
+  // index PG Granules into ES
+  await Promise.all(
```

Comment:
Yeah, bummer. Do we want to add any comments or move ES-related setup to a place that's easily fenced off and removed for the forward release?

Reply:
Hmm, yeah, probably that's a good QOL refactor for the main merge.    TODO.

Reply:
Check out 77c70b4974a